### ISSUE TITLE
lint: add normalization checks

### ISSUE THREAD

**Issue by Quasilyte** [go-critic/go-critic#21]
Don't know how to categorize changes, but we can detect several excessive AST elements.

```go
switch true {/* body */}
// =>
switch {/* body */}

// This case is handled by gosimple:
for true {/* body */}
// =>
for {/* body */}
```
There could be more.
We must implement only new checks that are not present in, say, [gosimple](https://github.com/dominikh/go-tools/tree/master/cmd/gosimple).

It can also be a good idea to report them in gosimple issue tracker.
It's strange that it handles `for` statement, but not `switch`.
Maybe there is a rationale for that.
